### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by secondsun**
#How to Test
* Follow the README
* Run the app
* Press all the buttons

### CODE REVIEW

File: AuthExamples/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# AGDroid Authentication Examples: HTTP Basic and Digest Authentication for Aerogear
+---------
+Author: Summers Pittman (secondsun)
+Level: Beginner
+Technologies: Java, Android
+Summary: A basic example of HTTP Authentication
+Target Product: -
+Product Versions: -
+Source: https://github.com/aerogear/aerogear-android-cookbook/tree/master/AuthExamples
+
```
Comment (danielpassos) [inline AuthExamples/README.md]: The old Makdown format problem. Need add 3 space in the end of the lines

File: AuthExamples/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,28 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion 21
+    buildToolsVersion "21.1.2"
+
+    defaultConfig {
+        applicationId "org.jboss.aerogear.authexamples"
```
Comment (danielpassos) [inline AuthExamples/app/build.gradle]: s/org.jboss.aerogear.authexamples/org.jboss.aerogear.android.cookbook.authexamples

File: AuthExamples/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (danielpassos) [inline AuthExamples/app/build.gradle]: I think it's not necessary

File: AuthExamples/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        <activity
+            android:name=".HowToUseHttpBasicAuthentication"
+            android:label="@string/app_name" />
+        <activity
+            android:name=".HowToUseDigestAuthentication"
+            android:label="@string/app_name" />
+        <activity
+            android:name=".MainActivity"
+            android:label="@string/title_activity_main" >
+            <action android:name="android.intent.action.MAIN" />
```
Comment (danielpassos) [inline AuthExamples/app/src/main/AndroidManifest.xml]: Missed `<intent-filter>`

File: AuthExamples/app/src/main/java/org/jboss/aerogear/authexamples/Constants.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.authexamples;
+
+public class Constants {
+    public static final String TAG = "AeroGear Cookbook";
```
Comment (danielpassos) [inline AuthExamples/app/src/main/java/org/jboss/aerogear/authexamples/Constants.java]: TAG is never used

File: AuthExamples/README.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+## Form Based Authenticator
+[AGReddit](../AGReddit) provides an example of custom Authentication using the Authentication libraries.
+
+## Notes
+There are two regressions to note for the 2.0.0 release with fixes scheduled for the 2.1.0 release.
+
+### Http Digest Authentication onLogout errors
+[This JIRA](https://issues.jboss.org/browse/AGDROID-349) is tracking the fix.  Digest Authentication log outs are not required to make a server call.  In cases where the server doesn't support a logout method (such as this) authentication will be cleared but the http request will fail.
+
+ ### Callbacks happen off the main thread
```
Comment (danielpassos) [inline AuthExamples/README.md]: Not sure why but markdown is not breaking line here

File: AuthExamples/README.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+### 1. Run Server
+
+```shell
+$ cd /path/to/authServer/
+$ mvn clean install wildfly:run
+```
+
+### 2. Configure Application
+
+Edit org.jboss.aerogear.authexamples.Constants#URL_BASE to be the path of the application.
```
Comment (danielpassos) [inline AuthExamples/README.md]: s/org.jboss.aerogear.authexamples.Constants#URL_BASE/`org.jboss.aerogear.authexamples.Constants#URL_BASE`

File: AuthExamples/README.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+$ gradle clean build
+```
+
+
+
+### 4. Test Application
+
+To deploy, run and debug the application on an Android device attached to your system, on the command line enter the following:
+
+2.1) Install generated apk to device
```
Comment (danielpassos) [inline AuthExamples/README.md]: s/2.1/4.1

File: AuthExamples/README.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+```shell
+$ cd /path/to/authExamples
+$ gradle installDebug
+```
+
+Now just run the app on your device/emulator
+
+## How does it work?
+
+```HotToUseHttpBasicAuthentication``` and ```HotToUseDigestAuthentication``` are two activities which create appropriate ```AuthenticationModules``` and associate them with ```Pipe``` objects.  The UI provides buttons to log in, log out, and fetch data from the Authentication Server.
```
Comment (danielpassos) [inline AuthExamples/README.md]: s/HotTo/HowTo

File: AuthExamples/README.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    AuthenticationModule module = authenticationConfig.asModule();
+//create the pipe
+//see HotToUseHttpBasicAuthentication.java#L91
+    RestfulPipeConfiguration pipeConfig = PipeManager.config("beer", RestfulPipeConfiguration.class)
+        .module(authModule)
+        .withUrl(new URL(Constants.URL_BASE + "/grocery/beers"));
+
+    Pipe<String> pipe = pipeConfig.forClass(String.class);
+///use the pipe as normal
+    pipe.read($callback);
```
Comment (danielpassos) [inline AuthExamples/README.md]: $?

File: AuthExamples/app/src/androidTest/java/org/jboss/aerogear/authexamples/ApplicationTest.java [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+import android.test.ApplicationTestCase;
+
+/**
+ * <a href="http://d.android.com/tools/testing/testing_android.html">Testing Fundamentals</a>
+ */
+public class ApplicationTest extends ApplicationTestCase<Application> {
+    public ApplicationTest() {
+        super(Application.class);
+    }
+}
```
Comment (danielpassos) [inline AuthExamples/app/src/androidTest/java/org/jboss/aerogear/authexamples/ApplicationTest.java]: :scream:

File: AuthExamples/app/src/main/res/layout/authentication.xml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+                android:id="@+id/logout"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_weight="1.0"
+                android:text="@string/logout"
+                android:enabled="false"/>
+
+    </LinearLayout>
+
+</LinearLayout>
```
Comment (danielpassos) [inline AuthExamples/app/src/main/res/layout/authentication.xml]: :scream:

File: AuthExamples/app/src/main/res/layout/authentication_with_data.xml [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+            android:id="@+id/logout"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_weight="1.0"
+            android:enabled="false"
+            android:text="@string/logout" />
+
+    </LinearLayout>
+
+</LinearLayout>
```
Comment (danielpassos) [inline AuthExamples/app/src/main/res/layout/authentication_with_data.xml]: :scream:

File: AuthExamples/app/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+    <string name="logout_successful">Logout successful</string>
+    <string name="logged">You are logged</string>
+    <string name="notlogged">You are *not* logged</string>
+    <string name="loading">Loading...</string>
+    <string name="how_to_use_http_basic_authentication">How To Use Httpbasic Authentication</string>
+    <string name="how_to_use_digest_authentication">How To Use Digest Authentication</string>
+    <string name="title_activity_main">Authentication Examples</string>
+    <string name="hello_world">Hello world!</string>
+    <string name="action_settings">Settings</string>
+
```
Comment (danielpassos) [inline AuthExamples/app/src/main/res/values/strings.xml]: `hello_world` and `action_settings` are not used

File: AuthExamples/gradle.properties [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+
+# Specifies the JVM arguments used for the daemon process.
+# The setting is particularly useful for tweaking memory settings.
+# Default value: -Xmx10248m -XX:MaxPermSize=256m
+# org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
+
+# When configured, Gradle will run in incubating parallel mode.
+# This option should only be used with decoupled projects. More details, visit
+# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
+# org.gradle.parallel=true
```
Comment (danielpassos) [inline AuthExamples/gradle.properties]: :scream:

File: AuthExamples/README.md (hunk 10 above)
Comment (lholmquist) [inline AuthExamples/README.md]: i think he's using jQuery here

File: AuthExamples/app/src/main/res/layout/authentication.xml (hunk 12 above)
Comment (secondsun) [inline AuthExamples/app/src/main/res/layout/authentication.xml]: What is :scream: ?

Comment (danielpassos) [inline AuthExamples/app/src/main/res/layout/authentication.xml]: :scream: Is "No newline at the end of file" :bowtie:

**Comment by secondsun** [comment]
@danielpassos Ok things should be taken care of now.

**Comment by danielpassos** [comment]
@secondsun I think git not `move` src file only `remove`. I can see the `app/src/main/java`

**Comment by danielpassos** [comment]
Awesome

**Pull request merged**